src/test/java/com/amazon/opendistroforelasticsearch/knn/plugin/action/RestKNNStatsHandlerIT.java

```diff
@@ -205,7 +207,8 @@ public void testScriptStats_singleShard() throws Exception {
                 StatNames.SCRIPT_QUERY_REQUESTS.getName())
         );
         nodeStats = parseNodeStatsResponse(EntityUtils.toString(response.getEntity()));
-        assertEquals(1, (int)(nodeStats.get(0).get(StatNames.SCRIPT_COMPILATIONS.getName())));
+        assertTrue((int)(nodeStats.get(0).get(StatNames.SCRIPT_COMPILATIONS.getName())) == 1 ||
+                (int)(nodeStats.get(0).get(StatNames.SCRIPT_COMPILATIONS.getName())) == initialScriptCompilations + 1);
```

Comment:
Why we need ```OR``` condition why not just 
```(int)(nodeStats.get(0).get(StatNames.SCRIPT_COMPILATIONS.getName())) == initialScriptCompilations + 1)```

Reply:
Yes, this was an attempted workaround. SCRIPT_COMPILATIONS will only be incremented if the script has yet to be compiled.

A better solution would be to clear the script cache before running the tests so that it is guaranteed that the script gets recompiled and SCRIPT_COMPILATIONS gets incremented by 1. I will update the PR.